org.eclipse.xtext/src/org/eclipse/xtext/xtext/ecoreInference/Xtext2EcoreTransformer.java

```diff
@@ -1016,12 +1019,20 @@ private EClassifierInfo createEClassifierInfo(TypeRef typeRef, String name) thro
 
 		EClassifier classifier = generatedEPackage.getEClassifier(classifierName);
 		if (classifier == null) {
-			if (GrammarUtil.containingParserRule(typeRef) != null)
+			if (GrammarUtil.containingParserRule(typeRef) != null) {
 				classifier = EcoreFactory.eINSTANCE.createEClass();
-			else if (GrammarUtil.containingEnumRule(typeRef) != null)
+			} else if (GrammarUtil.containingEnumRule(typeRef) != null) {
 				classifier = EcoreFactory.eINSTANCE.createEEnum();
-			else
-				throw new TransformationException(TransformationErrorCode.NoSuchTypeAvailable, "Cannot create datatype " + classifierName, typeRef);
+			} else {
+				for (AbstractMetamodelDeclaration mmd : grammar.getMetamodelDeclarations()) {
+					if (mmd instanceof ReferencedMetamodel) {
+						if (mmd.getEPackage() != null && mmd.getEPackage().getNsURI().equals(EcorePackage.eNS_URI)) {
```

Comment:
condition could be appended to preceding if condition

Reply:
upated